File: C++/섹션05. 완전 탐색 기본/02-01. BOJ 2606 - 바이러스 (DFS, 인접 행렬).cpp
#include <stdio.h>

#define MAX (100 + 10)

int V, E; // Vertex, Edge
int MAP[MAX][MAX];

bool visit[MAX];
int count;

void input()
{
	scanf("%d %d", &V, &E);

	for (int i = 0; i < E; i++)
	{
		int n1, n2;

		scanf("%d %d", &n1, &n2);

		MAP[n1][n2] = MAP[n2][n1] = 1;
	}
}

void printMap() // for debug
{
	for (int r = 1; r <= V; r++)
	{
		for (int c = 1; c <= V; c++)
			printf("%d ", MAP[r][c]);
		putchar('\n');
	}
	putchar('\n');
}

void DFS(int node)
{
	visit[node] = true;

	for (int c = 1; c <= V; c++)
	{
		if (MAP[node][c] == 0 || visit[c] == true) continue;

		count++;

		DFS(c);
	}
}

int main()
{
	input();

	DFS(1);

	printf("%d\n", count);

	return 0;
}

File: Python/섹션05. 완전 탐색 기본/02-01. BOJ 2606 - 바이러스 (DFS, 인접 행렬).py
MAX = 100 + 10

MAP = [[0] * MAX for _ in range(MAX)]
visit = [False] * MAX

count = 0

def input_data():
    global V, E

    V = int(input()) # Vertex
    E = int(input()) # Edge

    for _ in range(E):
        n1, n2 = map(int, input().split())
        MAP[n1][n2] = MAP[n2][n1] = 1    

def print_map(): # for debug
    for r in range(1, V + 1):
        print(*MAP[r][1:V + 1])

def DFS(node):      
    global count
      
    visit[node] = True

    for c in range(1, V + 1):        
        if MAP[node][c] == 0 or visit[c] == True: continue

        count = count + 1

        DFS(c)
       
input_data()
        
DFS(1)

print(count)
